src/python/pants/backend/project_info/paths.py

```diff
@@ -20,35 +20,24 @@
     TransitiveTargets,
     TransitiveTargetsRequest,
 )
+from pants.option.option_types import StrOption
 
 
 class PathsSubsystem(Outputting, GoalSubsystem):
     name = "paths"
     help = "List the paths between two addresses."
 
-    @classmethod
-    def register_options(cls, register):
-        super().register_options(register)
-        register(
-            "--from",
-            type=str,
-            dest="frm",
-            help="The path starting address",
-        )
-
-        register(
-            "--to",
-            type=str,
-            help="The path end address",
-        )
-
-    @property
-    def path_from(self) -> str:
-        return cast(str, self.options.frm)
+    path_from = StrOption(
+        "--from",
+        default=None,
+        help="The path starting address",
+    )
```

Comment:
:exclamation: This does have one minor semantic change. The ENV varm went from `PANTS_PATHS_FRM` to `PANTS_PATHS_FROM`. I doubt anyone is using the env var for this goal :sweat_smile: 

(The reason is this is the _only_ option with `dest`, and doesn't feel like a good enough reason to add it to the new system. I'm guessing it is this way because `self.options.from` would be a syntax error. Unfortunate that the internal attribute name leaked through to the ENV var name)